shared/js/event_dispatcher.js

```diff
@@ -101,6 +101,22 @@
     },
 
     /**
+     * Registers listener function to be executed only first time when event
+     * occurs.
+     * @param {string} eventName Name of the event to listen for.
+     * @param {function} handler Handler to be executed once event occurs.
+     */
+    once: function(eventName, handler) {
```

Comment:
I defined it to simplify patch, but I can switch back to ```on+off``` to get rid of shared peer review dependency :)